lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,174 @@
+;(function (define) {
+'use strict';
+
+define(['backbone',
+        'underscore',
+        'gettext',
+        'js/views/fields',
+        'teams/js/models/team',
+        'text!teams/templates/edit-team.underscore'],
+       function (Backbone, _, gettext, FieldViews, TeamModel, edit_team_template) {
+           return Backbone.View.extend({
+
+               maxTeamNameLength: 255,
+               maxTeamDescriptionLength: 300,
+
+               events: {
+                   "click .action-primary": "createTeam",
+                   "click .action-cancel": "cancelTeam"
+               },
+
+               initialize: function(options) {
+                   this.courseId = options.teamParams.courseId;
+                   this.teamsUrl = options.teamParams.teamsUrl;
+                   this.topicId = options.teamParams.topicId;
+                   this.languages = options.teamParams.languages;
+                   this.countries = options.teamParams.countries;
+
+                   _.bindAll(this, "cancelTeam", "createTeam");
+
+                   this.teamNameField = new FieldViews.TextFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Team Name (Required) *"),
+                       valueAttribute: 'name',
+                       bindEvents: false,
+                       helpMessage: gettext("A name that identifies your team (maximum 255 characters).")
+                   });
+
+                   this.teamDescriptionField = new FieldViews.TextareaFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Team Description (Required) *"),
+                       valueAttribute: 'description',
+                       editable: 'always',
+                       showMessages: false,
+                       bindEvents: false,
+                       descriptionMessage: gettext("A short description of the team to help other learners understand the goals or direction of the team (maximum 300 characters).")
+                   });
+
+                   this.optionalDescriptionField = new FieldViews.ReadonlyFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Optional Characteristics"),
+                       valueAttribute: 'optional_description',
+                       helpMessage: gettext("Help other learners decide whether to join your team by specifying some characteristics for your team. Choose carefully, because fewer people might be interested in joining your team if it seems too restrictive.")
```

Comment:
@muhammad-ammar Please add this sentence after "if it seems to restrictive.":
"You cannot change these characteristics after you create the team."